br/pkg/restore/data.go

```diff
@@ -0,0 +1,503 @@
+// Copyright 2022 PingCAP, Inc. Licensed under Apache-2.0.
+package restore
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"sort"
+	"time"
+
+	"github.com/emirpasic/gods/maps/treemap"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	recovpb "github.com/pingcap/kvproto/pkg/recoverdatapb"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/br/pkg/utils"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+)
+
+// recover the tikv cluster
+// 1. read all meta data from tikvs
+// 2. Assemble region meta and make recovery plan
+// 3. send the recover plan and the wait tikv to apply, in waitapply, all assigned region leader will check and apply log to the last log
+// 4. send the resolvedTs to tikv for deleting data.
+func RecoverData(ctx context.Context, resolvedTs uint64, allStores []*metapb.Store, tls *tls.Config, dumpRegionInfo bool) (int, error) {
+
+	numOfTiKVs := len(allStores)
```

Comment:
it seems we don't need this variable `numOfTiKVs`. change it by `len(allStores)` is enough?